### PR TITLE
Validate asynchrously (without sync support)

### PR DESCRIPTION
**Pull request by ylt**
Based on https://github.com/openactive/data-model-validator/pull/319

This is a major bump to 2.x.x as it's a breaking change.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,15 +67,17 @@ e.g.
 ```js
 const { validate } = require('@openactive/data-model-validator');
 
-const data = {
-  // ...
-};
+async function run() {
```
Comment (henryaddison) [inline README.md]: remove these wrapper "run" functions from documentation. I don't think it's useful to a developer.

**Comment by ylt** [comment]
Not entirely sure why Travis is failing here. "works for me"

```
➜  data-model-validator-async git:(ylt/validate-async) ✗ rm -rf node_modules
➜  data-model-validator-async git:(ylt/validate-async) ✗ git status
On branch ylt/validate-async
Untracked files:
  (use "git add <file>..." to include in what will be committed)

	data-model-validator-async.iml

nothing added to commit but untracked files present (use "git add" to track)
➜  data-model-validator-async git:(ylt/validate-async) ✗ nvm use 8
Now using node v8.10.0 (npm v5.6.0)
➜  data-model-validator-async git:(ylt/validate-async) ✗ npm install

> jsonpath@1.0.2 postinstall /Users/joecarter/projects/ODI/data-model-validator-async/node_modules/jsonpath
> node lib/aesprim.js > generated/aesprim-browser.js

added 243 packages in 5.839s
➜  data-model-validator-async git:(ylt/validate-async) ✗ npm test

> @openactive/data-model-validator@2.0.0 pretest /Users/joecarter/projects/ODI/data-model-validator-async
> npm run lint


> @openactive/data-model-validator@2.0.0 lint /Users/joecarter/projects/ODI/data-model-validator-async
> eslint "src/**/*.js"


> @openactive/data-model-validator@2.0.0 test /Users/joecarter/projects/ODI/data-model-validator-async
> npm run test-no-lint


> @openactive/data-model-validator@2.0.0 test-no-lint /Users/joecarter/projects/ODI/data-model-validator-async
> jasmine

Randomized with seed 42394
Started
....................................................................................................................................................................................................................................................................................................................................................


340 specs, 0 failures
Finished in 1.819 seconds
Randomized with seed 42394 (jasmine --random=true --seed=42394)
```

**Comment by ylt** [comment]
Definitely something up with travis 🤔 

This works also: `docker run --rm node:8 sh -c 'git clone https://github.com/openactive/data-model-validator.git --branch ylt/validate-async --single-branch; cd data-model-validator; npm install; npm test'`

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (henryaddison) [inline .vscode/launch.json]: Should this be checked in?

**Comment by henryaddison** [comment]
Yes curious about Travis - pr build fails, push build passes 🤷‍♂ . I hate npm.

**Pull request merged**